tfhe/src/core_crypto/algorithms/lwe_linear_algebra.rs

```diff
@@ -235,20 +279,51 @@ pub fn lwe_ciphertext_plaintext_add_assign<Scalar, InCont>(
 ) where
     Scalar: UnsignedInteger,
     InCont: ContainerMut<Element = Scalar>,
+{
+    let ciphertext_modulus = lhs.ciphertext_modulus();
+    if ciphertext_modulus.is_compatible_with_native_modulus() {
+        lwe_ciphertext_plaintext_add_assign_native_mod_compatible(lhs, rhs)
+    } else {
+        lwe_ciphertext_plaintext_add_assign_non_native_mod(lhs, rhs)
+    }
```

Comment:
As the three cases do different things, I would have 3 functions

Reply:
well we have 3 functions, this one dispatching, the compatible mod managing the native and power of 2 case, and the non native case

I don't know if having a function for each is better

Reply:
opinion on that @tmontaigu ? as well as the naming for the CiphertextModulusKind

Native => 2^n_bits for a n_bits integer (I think this one is fine)
NonNativePowerOfTwo => any 2^b for b != n_bits
NonNative => everything else

the last two are not too great

Reply:
`Native`,
`PowerOfTwo`,
`Custom`
?

Reply:
One advantage of having 3 separate functions is that it might be better for tight loop performances
```rust
match modulus_type {
Native => for bidule in machin { do_native(bidule) }
PowerOfTwo=> for bidule in machin { do_power_of_two(bidule) }
Custom=> for bidule in machin { do_custom(bidule) }
}
```

But since I dont `core` well, I don't know it that would actually be useful / measurable

Reply:
That's a fair point, though I think most loops would be in the underlying function anyway. It's just that I think we were fearing the growth in terms of number of lines of code. I'll take a second look, but generally speaking from a code point of view the power of 2 case is mapped to the native case via encryption, the few places it shows up are for plaintext encoding and removing noisy bits after the fft domain shenanigans. If the number of lines of code is not a cause for concern then I'll add specific variants in places where the code for native and power of 2 differ, if they are the same then I'm not a fan of duplicating the code as asserts would need to be adapted but the code would be essentially the same

Reply:
I don't have strong opinions on whether we should have 3 variants or not, unless we can measure a perf difference for some of them

Reply:
I think the NonNative will become Custom or Other, NonNativePowerOfTwo will stay as it describes well what it is